### DISCUSSION TITLE
MariaDB compatibility

### DISCUSSION THREAD

**Discussion by darkf3d3** [PrestaShop/PrestaShop#33098] [Q&A]
After the recent issue I encountered with the upgrade to prestashop 8.1, which would seem to be due to an incompatibility with the version of mariadb that I am using with prestashop 1.7 (MariaDB 10.1.48 - Specified key was too long; max key length is 767 bytes).
I searched for informations about prestashop version requirements, but sadly I couldn't find any.

How can we know which mariadb version to use?

We need something like this table made by wordpress:
![image](https://github.com/PrestaShop/PrestaShop/assets/42684711/c08cd7be-118c-4d1e-a41e-7b0224ab37c5)

**Comment by matks** [comment]
> We need something like this table made by wordpress:

Hi ! I'm going to provide a detailed answer

Writing the table is easy, what's hard is to make sure what you write the table is true.

To make sure "version X" of MySQL/MariaDB works with "version Y" of PrestaShop we would need to run our [nightly campaign test](https://nightly.prestashop-project.org/) with every version and verify it's green.

That would multiply the number of jobs we need to run every night and would be very expensive. We already run 8 1-hour campaign every night. Let's say we add tests for MySQL 5.6, MySQL 5.7 and MySQL 8.0 then it would need 8*3 = 24 jobs.

We would also need to verify different versions of MySQL when testing a new PrestaShop release, which is already very long (multiple full days of QA with multiple people for a patch release).

Not to mention the cost of maintaining these different usecases, when one test runs red, analyses what went wrong, deliver the bugfix,  test it...

If we had to expand our matrix of usecases I think we would first use different versions of PHP (just like we do in CI) before using MySQL.

<hr>

So what you ask for would be really useful and cool but it's very expensive (in terms of money (CPU is not free), dedicated maintainance time and complexity) so I think we can't do it.

If by any chance you're reaaaaally motivated 😄 and you want it to happen the code that runs the nighty test campaign is open source so you can reproduce and expand it on your own server 😉 .

**Comment by darkf3d3** [comment]
What I don't understand is what a marchant that want to install/upgrade prestashop should do.
Does he pick any version of mariadp and then cross his fingers hoping everything will work correctly?
In my case I'm stuck at PS1.7 because the upgrade to PS8.1 done using MariaDB 10.1.48 fail. And I don't know what to do because MariaDb version can't be easily switched like php, and after update it can't be downgraded.

**Comment by kpodemski** [comment]
@darkf3d3 

You can safely use MariaDB 10.10 or 10.11. I'm using 10.10 locally, and it works fine. I haven't tested MariaDB 11 yet.

**Comment by darkf3d3** [comment]
@kpodemski 
Thank you. I can use MariaDB 10.10 on both PS1.7.7 and PS8.1?
Because actually I'm still using PS1.7, but if MariaDB 10.10 it's safe to be used on it then I can proceed with the upgrade to PS8.1.

**Reply by kpodemski** [reply]
I didn't have any issues with MariaDB 10.10 and PrestaShop 1.7.7 :)

**Comment by matks** [comment]
See https://github.com/PrestaShop/PrestaShop/issues/33157#issuecomment-2090821496
